### DISCUSSION TITLE
A lot of features do not have any finite values. Filling with zeros.

### DISCUSSION THREAD

**Discussion by kiraminnie** [blue-yonder/tsfresh#1068] [Q&A]
Dear all, we are using tsfresh to analyse two kind of data: temporal series and spectrum. 
For both the data we are observing an unexpected behaviour.
Many features are evaluated as infinite and then filled with zeros.
This is strange because this happen fo two different kind of data, both with a range of value very regular (between -100 and +100 the first and the second between -4 and +4).
We tried to perform the features extraction using as input a standard signal, a sine wave, and we obtained the same results.
So, I need to understand why this happen.
Do you have some clues?

Thank you so much

Attached some screenshots of the features showing problems.

<img width="781" alt="Screenshot 2024-04-11 at 14 54 31" src="https://github.com/blue-yonder/tsfresh/assets/47897898/c3b26b78-1c2a-476d-bb22-c7fe1cde8761">
 
<img width="1586" alt="Screenshot 2024-04-11 at 14 53 54" src="https://github.com/blue-yonder/tsfresh/assets/47897898/d8301efa-9379-4378-ad1b-ebb7446b34e5">
<img width="569" alt="Screenshot 2024-04-11 at 14 54 04" src="https://github.com/blue-yonder/tsfresh/assets/47897898/bf902355-69eb-476c-ba48-f15c90100997">
<img width="348" alt="Screenshot 2024-04-11 at 14 54 13" src="https://github.com/blue-yonder/tsfresh/assets/47897898/f0330b95-6f26-4557-9056-700fb75d71a7">
<img width="523" alt="Screenshot 2024-04-11 at 14 54 23" src="https://github.com/blue-yonder/tsfresh/assets/47897898/58450700-1b51-44e5-97a4-ed81e2e57880">

**Comment by kempa-liehr** [comment]
Can you provide a short Python script, which reproduces the problem, e.g. for the sine data set?

**Comment by kiraminnie** [comment]
[test_sinus.pdf](https://github.com/blue-yonder/tsfresh/files/14997104/test_sinus.pdf)
[features_sinus.csv](https://github.com/blue-yonder/tsfresh/files/14997106/features_sinus.csv)

Thanks

**Comment by nils-braun** [comment]
Hi @kiraminnie! 
Unfortunately, I do not have your input data but only the features (where the issue has already happened and they are nan/inf). However from the "length" feature in your data I see, that your input data only had two rows. This is very short and most of our feature extractors can not extract anything meaningful from a timeseries with only two data points. This is most likely the reason for all the inf/nan features.
